src/controllers/user.controller.js

```diff
@@ -25,31 +26,23 @@ export default class UserController {
         preferred_currency: req.body.preferred_currency,
         location: req.body.location
       });
-
-      const userToken = auth.generateToken(user);
-      return res.status(201).json({
-        status: 'success',
-        data: {
-          id: user.id,
-          first_name: user.first_name,
-          last_name: user.last_name,
-          email: user.email,
-          role: user.role,
-          gender: user.gender,
-          birth_date: user.birth_date,
-          preferred_language: user.preferred_language,
-          preferred_currency: user.preferred_currency,
-          location: user.location,
-          isVerified: user.isVerified,
-          token: userToken
-        }
-      });
+      user.dataValues.token = UserUtils.generateToken(user);
+      return Responses.getSuccessResponse(res,
+        UserUtils.getPublicUserFields(user.dataValues), 201);
     } catch (error) {
-      return res.status(500)
-        .json({
-          status: 'error',
-          error: 'Could not signup user',
-        });
+      return Responses.getInternalServerErrorResponse(res, 'Could not signup user');
```

Comment:
First of all, did you refactor Oluchi's work?

Reply:
I did touch some of her files, mostly those common to my work as both are authentication routes. I did reach out to her for most of the changes and that is the main reason I assigned her as a reviewer on this PR

Reply:
Alright then